packages/flame_tiled/lib/src/renderable_tile_map.dart

```diff
@@ -81,98 +103,150 @@ class RenderableTiledMap {
+            return _RenderableImageLayer.load(layer as ImageLayer, camera);
+
+          default:
+            return _UnrenderableLayer.load(layer);
+        }
+      }),
     );
 
     return RenderableTiledMap(
       map,
-      batchesByLayer,
-      destTileSize,
+      renderableLayers,
     );
   }
 
-  static Iterable<TileLayer> _renderableTileLayers(TiledMap map) {
-    return map.layers.where((layer) => layer.visible).whereType<TileLayer>();
+  /// Handle game resize and propagate it to renderable layers
+  void handleResize(Vector2 canvasSize) {
+    renderableLayers.forEach((rl) {
+      rl.handleResize(canvasSize);
+    });
   }
 
-  static Future<Map<String, SpriteBatch>> _loadImages(TiledMap map) async {
-    final result = <String, SpriteBatch>{};
-
-    await Future.forEach(map.tiledImages(), (TiledImage img) async {
-      final src = img.source;
-      if (src != null) {
-        result[src] = await SpriteBatch.load(src);
-      }
+  /// Rebuilds the cache for rendering
+  void _refreshCache() {
+    renderableLayers.forEach((rl) {
+      rl.refreshCache();
     });
-
-    return result;
   }
 
-  /// Rebuilds the cache for rendering
-  void refreshCache() {
-    batchesByLayer.forEach(
-      (batchMap) => batchMap.values.forEach((batch) => batch.clear()),
-    );
+  /// Renders each renderable layer in the same order specified by the Tiled map
+  void render(Canvas c) {
+    if (_backgroundPaint != null) {
+      c.drawPaint(_backgroundPaint!);
+    }
 
-    _renderableTileLayers(map)
-        .where((e) => e.tileData != null)
-        .forEachIndexed((mapIndex, layer) {
-      return _renderLayer(
-        mapIndex,
-        layer.tileData!,
-        Vector2(layer.offsetX, layer.offsetY),
-      );
+    // paint each layer in reverse order, because the last layers should be
+    // rendered beneath the first layers
+    renderableLayers.where((l) => l.visible).forEach((renderableLayer) {
+      renderableLayer.render(c);
     });
   }
 
-  void _renderLayer(
-    int mapIndex,
-    List<List<Gid>> tileData,
-    Vector2 layerOffset,
+  /// Returns a layer of type [T] with given [name] from all the layers
+  /// of this map. If no such layer is found, null is returned.
+  T? getLayer<T extends Layer>(String name) {
+    return map.layers
+        .firstWhereOrNull((layer) => layer is T && layer.name == name) as T?;
+  }
+}
+
+abstract class _RenderableLayer<T extends Layer> {
+  final T layer;
+
+  _RenderableLayer(this.layer);
+
+  bool get visible => layer.visible;
+
+  void render(Canvas canvas);
+
+  void handleResize(Vector2 canvasSize) {}
+
+  void refreshCache() {}
+}
+
+class _RenderableTileLayer extends _RenderableLayer<TileLayer> {
+  final Camera? _camera;
+  final TiledMap _map;
+  final Vector2 _destTileSize;
+  late final _layerPaint = ui.Paint();
+  late final Map<String, SpriteBatch> _cachedSpriteBatches;
+
+  _RenderableTileLayer(
+    super.layer,
+    this._map,
+    this._camera,
```

Comment:
If I understand correctly, the only time you need access to camera is in `render()`. So in that case, you can modify `TiledComponent` to have `HasGameRef` mixin (similar to `ParallaxComponent`). This will allow you to access the camera via `gameRef.camera`. You can then modify the render methods of `RenderableTiledMap` and `_RenderableLayer` to accept an additional Camera parameter as input.

Reply:
That's a great point, would it be bad if it only worked for `FlameGame` and not `Game` though? (I think only `FlameGame` has `.camera`)

Reply:
I think you're right, making it work for `FlameGame` is sufficient. 

The existing version of this PR shouldn't be breaking though since it's an optional parameter at the entry point of `TiledComponent`, but I'll refactor like you're suggesting sometime this week if I have time

Reply:
I wound up setting defaulting the camera to `gameRef.camera` in `TiledComponent.gameResize` because it's needed during resize. This still allows client code to pass in their own camera if they want.

Reply:
Ah, good point, I was moving too fast when I was refactoring and didn't notice that I can just use `canvasSize` directly. I'll make that change, which will make it so `ImageLayer` will work without a `Camera`, and therefore `Camera` will only be necessary for parallax support